### PR TITLE
Add e2e test for the sign command

### LINKED ISSUE #4099: [E2E] Improve coverage for sign command
## Version of Singularity:

All.

### Expected behavior

All commands expected to succeed should be tested.
Part of issue #4078

### Actual behavior

- [ ] singularity sign --groupid
- [x] singularity sign --id
- [x] singularity sign --keyidx
- [ ] singularity sign --url

### Steps to reproduce behavior

N/A.

### PR DESCRIPTION
**Pull request by gvallee**
**Description of the Pull Request (PR):**

This PR adds E2E test to improve the coverage for the `sign` command. It only focuses of positive tests, not error cases.

**This fixes or addresses the following GitHub issues:**

- Addresses #4099 

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: e2e/sign/sign.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+			0,
+			e2e.ExpectOutput(e2e.ContainMatch, "Signature created and applied to "+imgPath),
+		),
+	)
+}
+
+func (c *ctx) generateKeypair(t *testing.T) {
+	keyGenInput := []e2e.SingularityConsoleOp{
+		e2e.ConsoleSendLine("e2e sign test key"),
+		e2e.ConsoleSendLine("geoffroy@sylabs.io"),
```
Comment (jscook2345) [inline e2e/sign/sign.go]: I would not use your own email here. Perhaps a `jdoe@sylabs.io`  or something else?

File: e2e/suite.go [hunk 2]
```diff
@@ -17,42 +17,27 @@ import (
 	"syscall"
```
Comment (jscook2345) [inline e2e/suite.go:17]: Perhaps some of this should be in a different PR, but I think it's OK to include now.

**Review by jscook2345 — CHANGES_REQUESTED** [review]
See comments

File: e2e/suite.go (hunk 2 above)
Comment (gvallee) [reply e2e/suite.go:17]: @jscook2345 either way works for me and it is there, it is really up to you :)

**Review by WestleyK — APPROVED** [review]
Looks good to me :+1: Thanks! Maybe at one point we should update this to run `verify` after the `sign`, just to make sure the image is actually signed. But that should be a update PR later, this is good as is.

File: e2e/suite.go (hunk 2 above)
Comment (gvallee) [reply e2e/suite.go:17]: I am lazy, let's keep it

**Comment by gvallee** [comment]
@WestleyK I opened issue #4229 to capture your comment about verifying the image. It is a very good point.

**Review by jscook2345 — APPROVED** [review]
